### PR TITLE
wallet-ext: welcome page wait for creating zklogin to finish

### PR DESCRIPTION
**Pull request by pchrysochoidis**
## Description 

* wait to finish before navigating to tokens page
* this fixes an issue where the account is locked and the unlocked (now it wait for it to unlock)
* this adds an enabled flag to init guard since this was that was doing the redirect
* creating zklogin account time will be improved with APPS-1621

before

https://github.com/MystenLabs/sui/assets/10210143/fda96339-db92-4148-a8d5-b44f13d731b0

after

https://github.com/MystenLabs/sui/assets/10210143/6822b16b-d7fe-4a77-84f2-5b7e341e9ac6

closes APPS-1501

## Test Plan 

👀 

---
If your changes are not user-facing and not a breaking change, you can skip the following section. Otherwise, please indicate what changed, and then add to the Release Notes section as highlighted during the release process.

### Type of Change (Check all that apply)

- [ ] protocol change
- [ ] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### CODE REVIEW

**Review by Jibz1 — APPROVED** [review]

**Review by mamos-mysten — APPROVED** [review]

**Comment by mamos-mysten** [comment]
as a side note - I'm finding the logic in `useInitializedGuard` a bit hard to follow, we can probably improve how we handle these redirects
